portal-impl/src/com/liferay/portal/spring/aop/ServiceBeanMethodInvocation.java

```diff
@@ -109,8 +108,8 @@ public void mark() {
 
 	@Override
 	public Object proceed() throws Throwable {
-		if (_index < _methodInterceptors.length) {
-			return _methodInterceptors[_index++].invoke(this);
+		if (_index < _chainableMethodAdvices.length) {
+			return _chainableMethodAdvices[_index++].invoke(this);
 		}
 
 		if (_equalsMethod) {
```

Comment:
One thing I was thinking about is we might want to pull out the equals logic as a separate MethodInterceptor.  That way we could do the equals checking logic once when we create the InvocationHandler and don't have to keep this logic here.

Reply:
Yes, that's on the to do list. But I am planning to make a special AopMethod for this, rather than a method interceptor for it. We pretty much never want to aop the equals method, we just want a special handling to use the inner target for equals testing. There are still a few small items to clean up before applying the bean post processor thing.